File: yep/src/main/java/catchla/yep/fragment/BlockedUsersFragment.java
package catchla.yep.fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.LoaderManager;
import android.support.v4.content.Loader;

import java.util.List;

import catchla.yep.adapter.UsersAdapter;
import catchla.yep.loader.BlockedUsersLoader;
import catchla.yep.model.TaskResponse;
import catchla.yep.model.User;
import catchla.yep.util.Utils;

/**
 * Created by mariotaku on 15/10/10.
 */
public class BlockedUsersFragment extends AbsContentRecyclerViewFragment<UsersAdapter>
        implements LoaderManager.LoaderCallbacks<TaskResponse<List<User>>> {

    @Override
    public void onRefresh() {
        super.onRefresh();
        final Bundle loaderArgs = getArguments();
        getLoaderManager().restartLoader(0, loaderArgs, this);
    }

    @Override
    public void onActivityCreated(@Nullable final Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        final Bundle loaderArgs = getArguments();
        getLoaderManager().initLoader(0, loaderArgs, this);
    }

    @Override
    public boolean isRefreshing() {
        return false;
    }

    @NonNull
    @Override
    protected UsersAdapter onCreateAdapter(Context context) {
        return new UsersAdapter(context);
    }

    @Override
    public Loader<TaskResponse<List<User>>> onCreateLoader(final int id, final Bundle args) {
        return new BlockedUsersLoader(getActivity(), Utils.getCurrentAccount(getActivity()));
    }

    @Override
    public void onLoadFinished(final Loader<TaskResponse<List<User>>> loader, final TaskResponse<List<User>> data) {
        getAdapter().setData(data.getData());
        showContent();
        setRefreshing(false);
    }

    @Override
    public void onLoaderReset(final Loader<TaskResponse<List<User>>> loader) {
        getAdapter().setData(null);
    }
}

File: src/main/kotlin/catchla/yep/fragment/BlockedUsersFragment.kt
package catchla.yep.fragment

import android.accounts.Account
import android.content.Context
import android.os.Bundle
import android.support.v4.app.LoaderManager
import android.support.v4.content.Loader
import catchla.yep.Constants
import catchla.yep.adapter.UsersAdapter
import catchla.yep.loader.BlockedUsersLoader
import catchla.yep.model.User

/**
 * Created by mariotaku on 15/10/10.
 */
class BlockedUsersFragment : AbsContentListRecyclerViewFragment<UsersAdapter>(), LoaderManager.LoaderCallbacks<List<User>?> {

    override fun onRefresh() {
        super.onRefresh()
        val loaderArgs = arguments
        loaderManager.restartLoader(0, loaderArgs, this)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val loaderArgs = arguments
        loaderManager.initLoader(0, loaderArgs, this)
    }

    override var refreshing: Boolean
        get() = false
        set(value) {
            super.refreshing = value
        }

    override fun onCreateAdapter(context: Context): UsersAdapter {
        return UsersAdapter(context)
    }

    val account: Account
        get() = arguments.getParcelable(Constants.EXTRA_ACCOUNT)

    override fun onCreateLoader(id: Int, args: Bundle?): Loader<List<User>?> {
        return BlockedUsersLoader(activity, account)
    }

    override fun onLoadFinished(loader: Loader<List<User>?>, data: List<User>?) {
        adapter.users = data
        showContent()
        refreshing = false
    }

    override fun onLoaderReset(loader: Loader<List<User>?>) {
        adapter.users = null
    }
}
